python/tests/test_adjoint_jax.py

```diff
@@ -16,7 +16,7 @@
 _FD_STEP = 1e-4
 
 # The tolerance for the adjoint and finite difference gradient comparison
-_TOL = 2e-2
+_TOL = 0.1 if mp.is_single_precision() else 0.02
```

Comment:
It makes sense that the error in finite-difference approximations would depend sensitively on the precision.   If you are computing [f(x+δ)-f(x)]/δ, the optimal choice of δ depends sensitively on the precision, and if δ is too small compared to the precision you can see a large "catastrophic cancellation" error.